### PR TITLE
Backport SameSite cookie attribute

### PR DESCRIPTION
**Pull request by mastahyeti**
Backports https://github.com/rack/rack/pull/980 and https://github.com/rack/rack/pull/1033 to the 1.6 branch.

/cc @jeremy @tenderlove

### CODE REVIEW

File: HISTORY.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+Sun Dec 4 18:48:03 2015  Jeremy Daer <jeremydaer@gmail.com>
+
+	* "First-Party" cookies. Browsers omit First-Party cookies from
+	third-party requests, closing the door on many common CSRF attacks.
+	Pass `first_party: true` to enable:
+	    response.set_cookie 'foo', value: 'bar', first_party: true
+
```
Comment (mastahyeti) [inline HISTORY.md]: Wasn't sure if this was supposed to be backported.

Comment (jeremy) [inline HISTORY.md]: Yeah, do omit. Cool to add your own entry though!

**Comment by mastahyeti** [comment]
Updated the HISTORY file. I just used the timestamp from 4d636d041a2466c860a9247edaa6aae736eca444.

**Comment by tenderlove** [comment]
Thanks!

**Pull request merged**